plaso/analyzers/hashers/entropy.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+"""The entropy calculation implementation."""
+
+from __future__ import division
+from __future__ import unicode_literals
+
+import collections
+import math
+
+from plaso.analyzers.hashers import interface
+from plaso.analyzers.hashers import manager
+
+
+class EntropyHasher(interface.BaseHasher):
+  """Calculates the byte entropy of input files."""
+
+  NAME = 'entropy'
+  DESCRIPTION = 'Calculates the entropy of input data.'
+
+  def __init__(self):
+    """Initializes the entropy hasher."""
+    super(EntropyHasher, self).__init__()
+    self._counter = collections.Counter()
+    self._length = 0
+
+  @classmethod
+  def GetAttributeName(cls):
+    """The attribute name for the result."""
+    return 'file_entropy'
+
+  def GetStringDigest(self):
+    """Calculates a unicode string containing the entropy value.
+
+    Byte entropy is a value between 0.0 and 8.0, and is returned as a string
+    to match the Plaso analyzer and storage APIs.
+
+    Returns:
+      str: byte entropy calculated over the data blocks passed to
+          Update().
+    """
+    if self._length == 0:
+      return '0.000000'
+
+    entropy = 0
+    for byte_frequency in self._counter.values():
+      byte_probability = byte_frequency / self._length
+      if byte_probability:
+        entropy += - byte_probability * math.log(byte_probability, 2)
+    return '{0:f}'.format(entropy)
```

Comment:
I opt to specify a fixed number of fraction digits here in the format string.

Reply:
Done